### PR TITLE
Slash Command Channel Parsing Bug Fixes

### LINKED ISSUE #1241: TypeError: 'tuple' object is not callable when a channel is provided as a slash command option. (2.0.0b7)
### Summary

Error in core.py when specifying a type of channel as a slash command option.

### Reproduction Steps

1. Create a slash command with an option parameter of type discord.TextChannel or discord.VoiceChannel.
2. Attempt to run the slash command and the error will occur.

### Minimal Reproducible Code

```python
@slash_command(guild_ids=globals.COMMAND_GUILD_ID_REGISTER_LIST)
async def testing(self, ctx, chan: Option(discord.TextChannel, "Specify channel:")):
    pass
```

### Expected Results

The code would run and I would have access to the channel object as the "chan" parameter, as per the example.

### Actual Results

Crash with traceback:

Traceback (most recent call last):
  File "c:\Users\Tiffany\dev\pycord-bot-testing\venv\lib\site-packages\discord\commands\core.py", line 122, in wrapped
    ret = await coro(arg)
  File "c:\Users\Tiffany\dev\pycord-bot-testing\venv\lib\site-packages\discord\commands\core.py", line 810, in _invoke
    arg = obj_type(state=ctx.interaction._state, data=_data, **kw)
TypeError: 'tuple' object is not callable

### Intents

All

### System Information

- Python v3.9.2-final
- py-cord v2.0.0-beta
    - py-cord pkg_resources: v2.0.0b7
- aiohttp v3.8.1
- system info: Windows 8.1 6.3.9600

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

Commands were previously working fine in any version under py-cord 2.0.0b6.

### PR DESCRIPTION
**Pull request by plun1331**
## Summary

- Fixes parsing of channels in slash command invocations:
```py
Traceback (most recent call last):
  File "C:\Users\cplun\Documents\GitHub\SkyKings Public Bot\venv\lib\site-packages\discord\commands\core.py", line 122, in wrapped
    ret = await coro(arg)
  File "C:\Users\cplun\Documents\GitHub\SkyKings Public Bot\venv\lib\site-packages\discord\commands\core.py", line 811, in _invoke
    arg = obj_type(state=ctx.interaction._state, data=_data, **kw)
TypeError: 'tuple' object is not callable
```
- Makes the `position` attribute optional because Discord wasn't sending it:
```py
Traceback (most recent call last):
  File "C:\Users\cplun\Documents\GitHub\SkyKings Public Bot\venv\lib\site-packages\discord\commands\core.py", line 122, in wrapped
    ret = await coro(arg)
  File "C:\Users\cplun\Documents\GitHub\SkyKings Public Bot\venv\lib\site-packages\discord\commands\core.py", line 810, in _invoke
    arg = obj_type(state=ctx.interaction._state, data=_data, **kw)
  File "C:\Users\cplun\Documents\GitHub\SkyKings Public Bot\venv\lib\site-packages\discord\channel.py", line 176, in __init__
    self._update(guild, data)
  File "C:\Users\cplun\Documents\GitHub\SkyKings Public Bot\venv\lib\site-packages\discord\channel.py", line 195, in _update
    self.position: int = data["position"]
KeyError: 'position'
```

There was apparently a PR for this (#1246) but it was closed

Fixes #1241

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

File: discord/channel.py [hunk 1]
```diff
@@ -130,9 +130,9 @@ class TextChannel(discord.abc.Messageable, discord.abc.GuildChannel, Hashable):
         The category channel ID this channel belongs to, if applicable.
     topic: Optional[:class:`str`]
         The channel's topic. ``None`` if it doesn't exist.
-    position: :class:`int`
+    position: Optional[:class:`int`]
         The position in the channel list. This is a number that starts at 0. e.g. the
-        top channel is position 0.
+        top channel is position 0. Can be ``None`` if the channel was recieved in an interaction.
```
Comment (plun1331) [inline discord/channel.py:135]: I don't know how true this comment is, it can be removed if needed

**Comment by jab416171** [comment]
does this fix #1241 and #1244?

**Comment by jab416171** [comment]
I'm just asking because you called out #1246 which fixes #1244.

**Comment by plun1331** [comment]
> I'm just asking because you called out #1246 which fixes #1244.

I don't think that PR fixed that ngl

File: discord/channel.py (hunk 1 above)
Comment (krittick) [reply discord/channel.py:135]: Official docs say that resolved interaction data returns partial channel objects and: 
> Partial Channel objects only have id, name, type and permissions fields.

So this is [expected to be] true.

**Comment by krittick** [comment]
LGTM

**Review by krittick — APPROVED** [review]

**Review by Dorukyum — APPROVED** [review]

**Comment by ResetXD** [comment]
issue still isnt resolved, the error still exists for slash command option of textchannel

**Comment by argo0n** [comment]
> issue still isnt resolved, the error still exists for slash command option of textchannel

Reinstalled Pycord through Github and it worked for me though

**Comment by ResetXD** [comment]
> > issue still isnt resolved, the error still exists for slash command option of textchannel
> 
> Reinstalled Pycord through Github and it worked for me though

weird cause it didnt for me...

**Pull request merged**